Factory Pattern/src/OrderEnum.java

```diff
@@ -0,0 +1,14 @@
+public enum OrderEnum {
+    FOOD("FoodOrder"),
```

Comment:
instead of string can use FoodOrder.class or can use `Supplier`. in case of string if the class name is changed this will break.